### ISSUE TITLE
Environment variables are not properly set

### ISSUE THREAD

**Issue by Permanently** [fehbari/sunshine-scripts#10]
When environment variables are set mid-execution, they are not passed through to the script.
![0frlSrEPuD](https://github.com/user-attachments/assets/6c6ac3c0-4384-4774-883c-4902152be165)

**Comment by fehbari** [comment]
On a somewhat related note to this bug, I've been thinking about how to best handle the stop script. Strictly speaking, it's not really necessary to set the resolution and refresh rate, as only the virtual display should've been changed when the session started. 

However, it can be useful to try and set the resolution in case something went wrong and we need to restore the main display's original state. But that can't really be done reliably as I don't really know how many displays one has, their arrangement, features and if you want the same settings for all of them. QRes will simply try to set this on the current main display, which may not be enough for multi-monitor setups, and might even do more harm than good.

Another issue is, it can get a bit more complicated to provide an easy to import `stop_streaming_task.xml` Task Scheduler task if people need to manually change up too many arguments after importing.

Things like HDR and FRL are safer to manipulate as they are applied globally by their respective CLI tools, so I'm leaning towards dropping the QRes changes from the `stop_streaming` script and only doing the things that are known to work more reliably - that is, disabling the virtual monitor, setting HDR (off by default), FRL and G-Sync.

But first I need to fix these custom arguments not being passed on correctly.

**Comment by Permanently** [comment]
When a resolution is set with the virtual monitor turned off, I tend to find my main monitor is adjusted. Speaking as someone with three physical monitors.

What may be worth is having a batch file, and then having a configuration file? That way, it'd mean one central place for adjustments. It'd also make adding extra options easy - I was actually looking at adding the option of changing C:\Tools to anything based on an env, which is how I found this out in the first place anyway. With scheduled tasks in mind as well, it only makes sense to have some sort of config.ini (or whatever it may be), to avoid tedious editing in multiple programs.

I'll do a branch tomorrow and see what I cook up, but for now election night's stealing my attention

**Comment by fehbari** [comment]
> When a resolution is set with the virtual monitor turned off, I tend to find my main monitor is adjusted

Yes, that's what QRes does in my experience if you don't specify a monitor. Perhaps there are newer tools out there with CLI interfaces that would allow for more flexibility in specifying the target monitor for changes.
 
> What may be worth is having a batch file, and then having a configuration file? That way, it'd mean one central place for adjustments. It'd also make adding extra options easy

That'd be a nice stepping stone towards my end goal with all this. If I had infinite time, I'd try to automate not just the setup and teardown of streaming sessions, but also the setup steps required by my scripts as well.

An _ideal_ flow in my mind would be, short of having a GUI wizard:

- Have a single interactive install script that populates your choices in a config file.
- Automate the process of downloading all the required tools and scripts to a folder of your choice (per the config).
- Download and install the virtual display driver. That'd be a good use case for `PnpUtil`.
    - As a bonus, set the virtual monitor as your main for games to use while it's enabled. Idk if that's possible only with PowerShell, might require additional tools or libraries.
- Add the start and stop scripts as _Do_ and _Undo_ commands in Sunshine settings, appending to any existing commands users might already have (or clearing them with their consent).
- Have the start and stop scripts load all their parameters from the config file (as you said yourself).

I believe that's all doable, but requires a decent amount of work and testing. For now, just doing the config file part would be a very good step in the right direction. 🙂

**Comment by Permanently** [comment]
> As a bonus, set the virtual monitor as your main for games to use while it's enabled. Idk if that's possible only with PowerShell, might require additional tools or libraries.

As much as I want this to be the case, it's hard to without at least AutoHotKey. Especially for, in example, PCSX2, where you can't actually specify what monitor to launch on. The workaround is having the monitor enabled, dragging the windows to that monitor, closing the program and then having it launch whenever the monitor's enabled again so it stays on the same display. Windows thing ¯\\\_(ツ)\_/¯

I don't think there'll be an easy solution for that. Probably possible, but 

> Have a single interactive install script that populates your choices in a config or env file.

This wouldn't actually be too hard once a config file is in place, it'll just be a case of having a configuration framework to work with.

**Comment by fehbari** [comment]
The original bug reported here should be fixed in [v1.2.0](https://github.com/fehbari/sunshine-scripts/releases/tag/1.2.0). Parameter reading has been fully reworked, so make sure to update your Sunshine "Do" and "Undo" commands.

It should be easier to test the scripts from the command line now, and the changes also lay a better foundation for loading default preferences from a config file later on.

I created https://github.com/fehbari/sunshine-scripts/issues/12 to track the bigger goal of automating initial setup and provide the centralized configuration discussed here. We can keep that conversation going there. 🙂
